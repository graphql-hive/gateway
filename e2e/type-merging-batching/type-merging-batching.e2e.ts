import { createTenv, Gateway } from '@internal/e2e';
import { beforeAll, expect, it } from 'vitest';

const { service, gateway } = createTenv(__dirname);

let gw: Gateway;
beforeAll(async () => {
  gw = await gateway({
    supergraph: {
      with: 'mesh',
      services: [await service('authors'), await service('books')],
    },
  });
});

it.each([
  {
    name: 'Author',
    query: /* GraphQL */ `
      query Author {
        author(id: 1) {
          id
          name
          books {
            id
            title
            author {
              id
              name
            }
          }
        }
      }
    `,
  },
  {
    name: 'Authors',
    query: /* GraphQL */ `
      query Authors {
        authors {
          id
          name
          books {
            id
            title
            author {
              id
              name
            }
          }
        }
      }
    `,
  },
  {
    name: 'Book',
    query: /* GraphQL */ `
      query Book {
        book(id: 1) {
          id
          title
          author {
            id
            name
            books {
              id
              title
            }
          }
        }
      }
    `,
  },
  {
    name: 'Books',
    query: /* GraphQL */ `
      query Books {
        books {
          id
          title
          author {
            id
            name
            books {
              id
              title
            }
          }
        }
      }
    `,
  },
])('should execute $name', async ({ query }) => {
  await expect(gw.execute({ query })).resolves.toMatchSnapshot();
});
