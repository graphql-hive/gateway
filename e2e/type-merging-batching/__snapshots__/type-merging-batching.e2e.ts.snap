// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should execute 'Books' 1`] = `
{
  "data": {
    "author": {
      "books": [
        {
          "author": {
            "id": "1",
            "name": "Jane Doe",
          },
          "id": "0",
          "title": "Lorem Ipsum",
        },
      ],
      "id": "1",
      "name": "Jane Doe",
    },
  },
}
`;

exports[`should execute 'Books' 2`] = `
{
  "data": {
    "authors": [
      {
        "books": [
          {
            "author": {
              "id": "0",
              "name": "John Doe",
            },
            "id": "1",
            "title": "Dolor Sit Amet",
          },
        ],
        "id": "0",
        "name": "John Doe",
      },
      {
        "books": [
          {
            "author": {
              "id": "1",
              "name": "Jane Doe",
            },
            "id": "0",
            "title": "Lorem Ipsum",
          },
        ],
        "id": "1",
        "name": "Jane Doe",
      },
    ],
  },
}
`;

exports[`should execute 'Books' 3`] = `
{
  "data": {
    "book": {
      "author": {
        "books": [
          {
            "id": "1",
            "title": "Dolor Sit Amet",
          },
        ],
        "id": "0",
        "name": "John Doe",
      },
      "id": "1",
      "title": "Dolor Sit Amet",
    },
  },
}
`;

exports[`should execute 'Books' 4`] = `
{
  "data": {
    "books": [
      {
        "author": {
          "books": [
            {
              "id": "0",
              "title": "Lorem Ipsum",
            },
          ],
          "id": "1",
          "name": "Jane Doe",
        },
        "id": "0",
        "title": "Lorem Ipsum",
      },
      {
        "author": {
          "books": [
            {
              "id": "1",
              "title": "Dolor Sit Amet",
            },
          ],
          "id": "0",
          "name": "John Doe",
        },
        "id": "1",
        "title": "Dolor Sit Amet",
      },
    ],
  },
}
`;
